### ISSUE TITLE
[ENH] Reduce docs build time

### ISSUE THREAD

**Issue by alexander-lakocy** [sktime/sktime#6772]
**Is your feature request related to a problem? Please describe.**
Documentation build takes a long time (>30 minutes for me) using `readthedocs` through the PR interface. It also takes ~15 minutes to build the docs on a `venv` on my machine.

**Describe the solution you'd like**
I would like to reduce the document build time to something more reasonable.

**Describe alternatives you've considered**
Continue waiting 30 minutes each time a PR is started.

**Additional context**
At the time of posting, doc builds using `readthedocs` are also failing due to memory issues. I am not sure how to resolve this.

**Comment by fkiraly** [comment]
Yes, I've also obsereved this - perhaps profiling can give insights where/what the compute usage is coming from

**Comment by yarnabrina** [comment]
It seems to affect #6841 too, and as per RTD docs, there are specific limits on timeout based on the community/business.

I do not have an explanation what can increase the build time significantly suddenly for specific PR's. @duydl do you have any suggestions?

**Comment by duydl** [comment]
Hi. I think RTD build has always been taking consistently more than 20 minutes, so maybe some random fluctuation could push it past the limit, which I believe is 15min for community plan and 30 for business. 

I could try suggesting something though not sure applicable.
- Disable/relax some style checks, particularly `numpydoc` : This check gives thousands of errors when building. They have quite heavy footprint also based on past profile. If the error could not be addressed maybe we could just disable the extension. A pre-commit would be more economic if there is some specific check wanted later. 
- Move some rendering to pre-commit etc. Like the estimator table could be done in other stages instead of when building doc. 
- I believe the notebooks aren't supposed to run during doc build, but just converted to html because no other extra dependencies are installed. But if i am not wrong the `jupyter` package is used to run nb. So maybe check if there any redundant code and if  `jupyter` could be removed ...

**Comment by yarnabrina** [comment]
> I believe the notebooks aren't supposed to run during doc build, but just converted to html because no other extra dependencies are installed. But if i am not wrong the jupyter package is used to run nb. So maybe check if there any redundant code and if jupyter could be removed from docs dependencies...

This makes sense to me, especially because we already run notebooks as part of normal CI. I did not know it was run as part of build, so we may remove it. I do not know how is it configured or how to disable that though, do you have any idea? Will it be possible for you to send a PR, if you have some time?

> Disable/relax some style checks, particularly numpydoc : This check gives thousands of errors when building. They have quite heavy footprint also based on past profile. If the error could not be addressed maybe we could just disable the extension. A pre-commit would be more economic if there is some specific check wanted later.

@fkiraly are you okay with disabling numpydoc? We are not monitoring those checks anyway.

**Comment by fkiraly** [comment]
> @fkiraly are you okay with disabling numpydoc? We are not monitoring those checks anyway.

I was not aware we were running these in the doctests, so I would not mind.
